.gitignore

```diff
@@ -25,3 +25,32 @@ _testmain.go
 
 vendor
 .cover
+
+# MacOS
+
+*.DS_Store
+.AppleDouble
+.LSOverride
+
+# Icon must end with two \r
+Icon
+
```

Comment:
I know your point but this is the `golang` project, not macOS project. We don't need to include all macOS ignore list.

If I merge this PR. somebody want to add some ignore items like https://github.com/github/gitignore/tree/master/Global.